libs/pageserver_api/src/keyspace.rs

```diff
@@ -124,6 +124,9 @@ impl KeySpaceAccum {
                 if range.start == accum.end {
                     accum.end = range.end;
                 } else {
+                    // TODO: to efficiently support small sharding stripe sizes, we should avoid starting
```

Comment:
May be I missed something, but it seems to me that for huge stripes excluding from key space ranges not belonging to the shard may be even more critical.